sdk/storage/azure-storage-blob-changefeed/tests/test_change_feed.py

```diff
@@ -97,12 +100,121 @@ def test_get_change_feed_events_with_continuation_token(self, resource_group, lo
     @GlobalStorageAccountPreparer()
     def test_get_change_feed_events_in_a_time_range(self, resource_group, location, storage_account, storage_account_key):
         cf_client = ChangeFeedClient(self.account_url(storage_account, "blob"), storage_account_key)
-        start_time = datetime(2020, 5, 12)
-        end_time = datetime(2020, 5, 13)
+        start_time = datetime(2020, 8, 12)
+        end_time = datetime(2020, 8, 18)
         change_feed = cf_client.list_changes(start_time=start_time, end_time=end_time, results_per_page=2).by_page()
 
         # print first page of events
         page1 = next(change_feed)
         events = list(page1)
 
         self.assertIsNot(len(events), 0)
+
+    @GlobalStorageAccountPreparer()
+    def test_read_change_feed_without_any_seg(self, resource_group, location, storage_account, storage_account_key):
+        cf_client = ChangeFeedClient(self.account_url(storage_account, "blob"), storage_account_key)
+        start_time = datetime(2021, 8, 19)
+        change_feed = cf_client.list_changes(start_time=start_time)
+
+        events = list(change_feed)
+        self.assertEqual(len(events), 0)
+
+    @GlobalStorageAccountPreparer()
+    def test_read_change_feed_tail_where_3_shards_have_data(self, resource_group, location, storage_account, storage_account_key):
+        cf_client = ChangeFeedClient(self.account_url(storage_account, "blob"), storage_account_key)
+
+        # to read until the end
+        start_time = datetime(2020, 8, 19, 23)
+        change_feed = cf_client.list_changes(start_time=start_time).by_page()
+
+        for page in change_feed:
+            for event in page:
+                print(event)
+        token = change_feed.continuation_token
+
+        self.assertEqual(token['CursorVersion'], 1)
+        self.assertIsNotNone(token['UrlHost'])
+        self.assertEqual(len(token['CurrentSegmentCursor']['ShardCursors']), 3)
+        self.assertIsNotNone(token['CurrentSegmentCursor']['SegmentPath'])
+        self.assertIsNotNone(token['CurrentSegmentCursor']['CurrentShardPath'])
+
+        if self.is_live:
+            sleep(120)
```

Comment:
In Java and C# we made these tests playback-only. is it a possibility in python?